packages/web/src/pages/track-page/components/mobile/TrackHeader.tsx

```diff
@@ -310,10 +373,27 @@ const TrackHeader = ({
         />
       </div>
       <div className={styles.buttonSection}>
-        <PlayButton playing={isPlaying} onPlay={onPlay} />
+        {isPremiumContentEnabled &&
+          !doesUserHaveAccess &&
```

Comment:
starting to feel like there's a lot of the same check throughout this pr - `isPremiumContentEnabled && !doesUserHaveAccess` or its inverse. Maybe `doesUserHaveAccess` could take into account the feature flag too to simplify some of the boolean logic (or make a third boolean that includes both)?

Reply:
100% agreed. it's in my plate to refactor this :)